src/drivers/dma/bcm283x/bcm283x_dma.c

```diff
@@ -0,0 +1,222 @@
+/**
+ * @file
+ * @brief
+ *
+ * @date    07.06.2021
+ * @author  kpishere
+ */
+
+#include <assert.h>
+#include <kernel/printk.h>
+#include <kernel/irq.h>
+#include <embox/unit.h>
+#include <framework/mod/options.h>
+#include <drivers/dma/dma.h>
+#include <module/embox/mem/bitmask.h>
+
+#include <asm/cp15.h>
+#include <kernel/printk.h>
+
+#include <drivers/bcm283x_dma.h>
+#include <drivers/mailbox/bcm2835_mailbox_property.h>
+
+#define DMA_0_BASE OPTION_GET(NUMBER, dma0_base)
+#define DMA_F_BASE OPTION_GET(NUMBER, dmaF_base)
+
+#define DMA330_IRQ OPTION_GET(NUMBER,dma_interrupt_irq)
+
+typedef struct {
+    volatile uint32_t cs;
+    volatile uint32_t conblk_ad;
+    volatile uint32_t ti;
+    volatile uint32_t source_ad;
+    volatile uint32_t dest_ad;
+    volatile uint32_t txfr_len;
+    volatile uint32_t stride;
+    volatile uint32_t nextconbk;
+    volatile uint32_t debug;
+} Dma;
+
+#define REGS_DMA(c) (( Dma *)( DMA_0_BASE + 0x100 * (c) ) )
+#define REGS_DMA_F  (( Dma *)( DMA_F_BASE ))
+
+#define REGS_DMA_STATUS ((volatile uint32_t *)( DMA_0_BASE + 0x0fe0))
+#define REGS_DMA_ENABLE ((volatile uint32_t *)( DMA_0_BASE + 0x0ff0))
+
+static volatile Dma_conbk shared_conbk; 
+
+#if 0
+static irq_return_t bcm283x_dma_irq_handler(unsigned int irq_num, void *dev_id) {
+//	HAL_DMA_IRQHandler(&dma_handle);
+
+	return IRQ_HANDLED;
+}
+STATIC_IRQ_ATTACH(DMA330_IRQ, bcm283x_dma_irq_handler, NULL);
+#endif
+
+
+// Returns a handle to the allocated memory
+//
+uint32_t vc_malloc(size_t bytes, uint32_t align, uint32_t flags) {
+    bcm2835_mailbox_property_t *resp;
+    
+    bcm2835_property_init();
+    bcm2835_property_add_tag( TAG_ALLOCATE_MEMORY,  bytes, align, flags );
+    bcm2835_property_process();
+    resp = bcm2835_property_get( TAG_ALLOCATE_MEMORY );
+    uint32_t handle = (uint32_t)resp->data.value_32;
+
+    return handle;
+}
+
+// Lock memory to physical address mapping, returns a bus address 
+// accessable only by GPU and DMA (but corresponding physical address
+// can now be accessed by ARM CPU)
+//
+uint32_t vc_mem_lock(uint32_t handle) {
```

Comment:
static ?

Reply:
I think so, only want to expose dma_malloc() and dma_free(), I'm not seeing any other need to make this kind of mailbox request.  Will be static too.